src/components/Elements.tsx

```diff
@@ -101,74 +87,49 @@ interface PrivateElementsProps {
  */
 export const Elements: FunctionComponent<ElementsProps> = ({
   stripe: rawStripeProp,
-  options,
+  options: optionsProp,
   children,
 }: PrivateElementsProps) => {
-  const final = React.useRef(false);
-  const isMounted = React.useRef(true);
-  const parsed = React.useMemo(() => parseStripeProp(rawStripeProp), [
-    rawStripeProp,
-  ]);
-  const [ctx, setContext] = React.useState<ElementsContextValue>(() => ({
-    stripe: null,
-    elements: null,
-  }));
-
-  const prevStripe = usePrevious(rawStripeProp);
-  const prevOptions = usePrevious(options);
-  if (prevStripe !== null) {
-    if (prevStripe !== rawStripeProp) {
+  const [inputs, setInputs] = React.useState({ rawStripe: rawStripeProp, options: optionsProp })
+  const { rawStripe, options } = inputs
```

Comment:
nit: I find it confusing to remember the difference between `rawStripe`/`rawStripeProp`, `options/optionsProp` etc. How about we:

- Remove the destructuring of the `Elements` function parameter and just name that parameter `props`
- Rename the `[inputs, setInputs]` state variables to `[savedProps, setSavedProps]`
- Refer to the various values everywhere as `props.stripe` / `savedProps.stripe`, `props.options` / `savedProps.options`, etc.

What do you think? Does that seem clearer?

Reply:
Totally agree with you! I thought it was so customary 😁.  But IMO it's better to name `inputs`, because it's some kind of input to construct context and if it's named `props` it should contain `children` as well.